parsers/robot.c

```diff
@@ -16,12 +16,16 @@
 #include "routines.h"
 
 typedef enum {
+        K_TESTCASE,
         K_KEYWORD,
         K_VARIABLE,
         COUNT_KIND
 } RobotKind;
 
+static RobotKind section = -1;
+
```

Comment:
It seems that this variable is initialized only here.
Though it is a bit verbose, this variable should be initialized each time when new input is given to this parser.
  findRobotTags() is the good place where `section` should be initialized.